workflow/executor/resource.go

```diff
@@ -247,6 +247,11 @@ func checkResourceState(resourceNamespace string, resourceName string, successRe
 			if err != nil {
 				log.Warnf("cmd.Wait for kubectl get -w command for resource %s returned error %v",
 					resourceName, err)
+
+				if checkIfResourceDeleted(resourceName, resourceNamespace) {
+					return false, errors.Errorf(errors.CodeNotFound, "Resource %s in namespace %s has been deleted somehow.", resourceName, resourceNamespace)
+				}
```

Comment:
I'm not too familiar with this bit of code and I'm not sure this is necessary, can you explain why you would like this change?

Reply:
This is necessary in our situation where resource has been deleted but the status doesn't get reflected since there's also an error from `cmd.Wait()`. The existing code only checks whether the resource has been deleted when there isn't error.

Reply:
Would moving `if checkIfResourceDeleted(resourceName, resourceNamespace) {` to the top of the `for` loop (i.e. before `jsonBytes, err := readJSON(reader)`) also helps solve this issue? If so, I think this solution would be preferable.

Reply:
I'll defer that to @merlintang

Reply:
thanks for your comments, I have updated this. please check it again.